### PR TITLE
Refactor docs with `@mtkmodel`

### PR DESCRIPTION
**Pull request by ven-k**
- Refactors "Getting Started" page to use `@mtkmodel` and adds a note on using `complete`
- Adds "Components" and "Connectors" section to MTKModel-Connector docs-page.
- And elaborates what `Model.structure` is.
- Refactors following tutorials with `@mtkmodel`
    - Acausal components 
    - Structural identifiability
- Adds `connect` section in Variable metadata
- Updates ContextualVariables.md with `get_unit` and `VariableUnit`
- Fixes u0 in Optimization tutorial

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
What's left to merge?

**Comment by ArnoStrouwen** [comment]
Could you retrigger CI @ChrisRackauckas?
I can at least resolve locally now.

**Comment by ChrisRackauckas** [comment]
Needs to be resolved to trigger.

File: docs/src/tutorials/parameter_identifiability.md [hunk 1]
```diff
@@ -31,33 +31,39 @@ We first define the parameters, variables, differential equations and the output
...
+    @variables begin
+        x4(t); x5(t); x6(t); x7(t)
+        y1(t), [output = true]
+        y2(t), [output = true]
+    end
+    @parameters begin
+        k5; k6; k7; k8; k9; k10
+    end
+    begin
+        D = Differential(t)
```
Comment (ArnoStrouwen) [inline docs/src/tutorials/parameter_identifiability.md:44]:
```suggestion
        @variables t
        D = Differential(t)
```

File: docs/src/tutorials/parameter_identifiability.md [hunk 2]
```diff
@@ -66,8 +72,7 @@ Let's try to check specific parameters and their combinations
 
 ```@example SI
 to_check = [k5, k7, k10 / k9, k5 + k6]
```
Comment (ArnoStrouwen) [inline docs/src/tutorials/parameter_identifiability.md:74]:
```suggestion
@unpack k5, k6, k7, k9, k10 = de
to_check = [k5, k7, k10 / k9, k5 + k6]
```

**Review by ArnoStrouwen — CHANGES_REQUESTED** [review]
Fix for the SI tutorial.

**Comment by ven-k** [comment]
> What's left to merge?

I'm refactoring the "Getting started" based on Friday's discussion. I'm planning to make a copy of it with `@mtkmodel` and rename the current "Getting started" as "Introduction to non-DSL based ModelingToolkit" (with maybe a little bit of minor cuts).

File: docs/src/tutorials/parameter_identifiability.md (hunk 1 above)
Comment (ven-k) [reply docs/src/tutorials/parameter_identifiability.md:44]: Will resolve the conflicts and commit this in.

**Comment by ArnoStrouwen** [comment]
I think the `@connector` macro needs a similar thorough explanation as the `@mtkmodel` macro is getting
Specifically, where Flow comes from.

Having docstrigns for both would also be a plus.

**Comment by ven-k** [comment]
Note that these have changed since the previous push:

- Refactors "Getting Started" page to use `@mtkmodel` and adds a note on using `complete`
- Adds "Components" and "Connectors" section to MTKModel-Connector docs-page.
- And elaborates what `Model.structure` is.
- Adds `connect` section in Variable metadata
- Updates ContextualVariables.md with `get_unit` and `VariableUnit`

**Comment by YingboMa** [comment]
What does it take to fix the Documentation CI?

**Comment by ven-k** [comment]
- https://github.com/SciML/StructuralIdentifiability.jl/issues/216

**Comment by ChrisRackauckas** [comment]
what's left to merge?

**Comment by ven-k** [comment]
I'll retrigger docs build.

**Comment by ChrisRackauckas** [comment]
Let's merge and follow up with more changes.

**Pull request merged**